### ISSUE TITLE
Intermittent test failures

### ISSUE THREAD

**Issue by dantswain** [pinterest/elixir-thrift#235]
Travis tests fail maybe 30% of the time (guesstimate) and it's always some kind of timeout that goes away with a rebuild.  @scohen I _think_ this is in the service / behaviour code and smells like a synchronization problem.  If you have some idea what might be failing, I have some tricks up my sleeve to deal with that kind of thing.

**Comment by fishcakez** [comment]
Travis looks in good shape? If this is still an issue can you show the failed builds please @dantswain?

**Comment by jparise** [comment]
I still see this occur from time-to-time on Travis. 30% still feels accurate.

**Comment by fishcakez** [comment]
@jparise are we restarting the build and hiding the failures? The travis UI makes this awkward to see the failures then, can you post output here next time it occurs here please.

**Comment by jparise** [comment]
@fishcake that's right, and I'll dump some more details here the next time it occurs.

**Comment by jparise** [comment]
From https://travis-ci.org/pinterest/elixir-thrift/jobs/283250924:

```
  1) test it returns :ok on void oneway functions if the server dies (Thrift.Generator.ServiceTest)
     test/thrift/generator/service_test.exs:416
     No message matching {:EXIT, _, {:error, :econnrefused}} after 100ms.
     Process mailbox:
       {:EXIT, #PID<0.2488.0>, :normal}
       {:EXIT, #PID<0.2490.0>, {:error, :econnreset}}
     code: assert_receive {:EXIT, _, {:error, :econnrefused}}, 100
     stacktrace:
       test/thrift/generator/service_test.exs:422: (test)
```
